Add support for app extensions

PatrickNLT opened the pull request:
WYPopover currently doesn't compile on iOS extensions.

A workaround is to wrap every forbidden call into a `#if !defined(WY_APP_EXTENSIONS)` macro. This is used in most libraries.

I've been using this fix in production for a few weeks already, it works very well.

Commit:
Add support for app extensions

PatrickNLT commented:
Would you consider merging this pull request?

vitalys commented:
@PatrickNLT I'm worried about using `UIInterfaceOrientationPortrait` for orientation in case of an app extension. How does it work on iPad?

Commit:
Add support for extensions in the podspec

Extension targets that require WYPopoverController should add a reference in their Podfile to 'WYPopoverController/AppExtension', not 'WYPopoverController'
